Gems/Automation/Code/Source/AutomationScriptBindings.cpp

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AutomationSystemComponent.h>
+
+#include <AutomationScriptBindings.h>
+
+#include <AzCore/Console/IConsole.h>
+#include <AzCore/Math/MathReflection.h>
+#include <AzCore/RTTI/BehaviorContext.h>
+
+#include <AzFramework/Components/ConsoleBus.h>
+
+namespace Automation
+{
+    namespace Bindings
+    {
+        void Print(const AZStd::string& message)
+        {
+            auto func = [message]()
+            {
+                AZ_TracePrintf("Automation", "Script: %s\n", message.c_str());
+            };
+
+            AutomationInterface::Get()->QueueScriptOperation(AZStd::move(func));
+        }
+
+        void Warning(const AZStd::string& message)
+        {
+            auto func = [message]()
+            {
+                AZ_Warning("Automation", false, "Script: %s", message.c_str());
+            };
+
+            AutomationInterface::Get()->QueueScriptOperation(AZStd::move(func));
+        }
+
+        void Error(const AZStd::string& message)
+        {
+            auto func = [message]()
+            {
+                AZ_Error("Automation", false, "Script: %s", message.c_str());
+            };
+
+            AutomationInterface::Get()->QueueScriptOperation(AZStd::move(func));
+        }
+
+        void ExecuteConsoleCommand(const AZStd::string& command)
+        {
+            auto func = [command]()
+            {
+                AzFramework::ConsoleRequestBus::Broadcast(
+                    &AzFramework::ConsoleRequests::ExecuteConsoleCommand, command.c_str());
+            };
+
+            AutomationInterface::Get()->QueueScriptOperation(AZStd::move(func));
+        }
+
+        void IdleFrames(int numFrames)
+        {
+            auto func = [numFrames]()
+            {
+                auto automationInterface = AutomationInterface::Get();
+                auto automationComponent = azrtti_cast<AutomationSystemComponent*>(automationInterface);
+
+                automationComponent->SetIdleFrames(numFrames);
+            };
+
+            AutomationInterface::Get()->QueueScriptOperation(AZStd::move(func));
+        }
+
+        void IdleSeconds(float numSeconds)
+        {
+            auto func = [numSeconds]()
+            {
+                auto automationInterface = AutomationInterface::Get();
+                auto automationComponent = azrtti_cast<AutomationSystemComponent*>(automationInterface);
+
+                automationComponent->SetIdleSeconds(numSeconds);
+            };
+
+            AutomationInterface::Get()->QueueScriptOperation(AZStd::move(func));
+        }
+    } // namespace Bindings
+
+    void ReflectScriptBindings(AZ::BehaviorContext* behaviorContext)
+    {
+        AZ::MathReflect(behaviorContext);
```

Comment:
Why is math reflection occuring in this function?

Reply:
I understand now.
The Automation gem is using it's own private BehaviorContext.